edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/authentication/AuthAgentProtocolHead.cs

```diff
@@ -66,15 +68,15 @@ public async Task StartAsync()
 
             await this.host.Expect(() => new Exception("No AUTH host instance found to start"))
                            .StartAsync();
-
+            this.startSignal.Set();
             Events.Started();
         }
 
         public async Task CloseAsync(CancellationToken token)
         {
             Events.Closing();
-
             Option<IWebHost> hostToStop;
+            this.startSignal.Reset();
```

Comment:
>this.startSignal.Reset(); [](start = 12, length = 25)

this should probably come after the call to StopAsync, otherwise there's a window of inconsistency.  Not an issue right now but could lead to bugs in the future.

Reply:
Will do. Nice catch.